File: makerNexts/pages/api/CopyTo.ts
// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
import type { NextApiRequest, NextApiResponse } from "next";

type Data = {
  code: number;
  message: string;
  data: any;
};

const sendRequest: (data: {
  sourcePath: string;
  destinationPath: string;
}) => Promise<any> = async ({ sourcePath, destinationPath }) => {
  const url = "http://127.0.0.1:8000/copyTo"; // 请求的 URL

  try {
    const response = await fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ sourcePath, destinationPath }),
    });

    if (!response.ok) {
      throw new Error("网络响应不正常，状态码：" + response.status);
    }

    return await response.json();
  } catch (error) {
    console.error("发生错误：", error);

    return { code: 500, message: error, data: null };
  }
};

export default async function Handler(
  req: NextApiRequest,
  res: NextApiResponse<Data>,
) {
  if (req.method === "POST") {
    const { sourcePath, destinationPath } = req.body;

    const result = await sendRequest({ sourcePath, destinationPath });

    return res.status(200).json(result);
  }
}

File: maker/api/copy_to/v1/copy_to.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

type Req struct {
	g.Meta          `path:"/copyTo" tags:"copyTo" method:"post" summary:"将源文件夹中的文件拷贝到目标文件夹中"`
	SourcePath      string `json:"sourcePath" validate:"required" example:"/path/to/source"`
	DestinationPath string `json:"destinationPath" validate:"required" example:"/path/to/destination"`
}
type Res struct {
	g.Meta  `mime:"application/json" example:"string"`
	Code    int    `json:"code"`
	Message string `json:"message"`
}
